### PR TITLE
GH-1247 - Upgrade doc updates

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1247.

Description:
- Do yum repolist after adding IML 5.1 repo to manager. Do this to confirm old IML repos have been terminated (only should have 5.1 repo).
- Remove all reboots on manager updates.
- iml update-repo <NODE_NAME_HERE> should be run for each host after upgrading manager.
- Add 5.0 to 5.1 upgrade doc

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

File: docs/Contributor_Docs/cd_Running_IML_On_Docker_With_Vagrant_Servers.md [hunk 1]
```diff
@@ -43,10 +43,10 @@ vagrant up iscsi mds1 mds2 oss1 oss2
 Verify that an ssh connection can be established to each of the server nodes from the host:
 
 ```sh
- ssh -i ~/repos/Vagrantfiles/iml-sandbox/id_rsa root@10.73.10.11 "hostname"
```
Comment (jgrund) [inline docs/Contributor_Docs/cd_Running_IML_On_Docker_With_Vagrant_Servers.md]: Can this be in a separate patch?

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 2]
```diff
@@ -61,19 +61,29 @@ The software upgrade process requires super-user privileges to run. Login as the
    yum-config-manager --add-repo=https://github.com/whamcloud/integrated-manager-for-lustre/releases/download/v5.1.0/chroma_support.repo
    ```
 
+1. Verify that the old iml-4.0.x repo file has been removed from the repolist and that the 5.1 repo has been added.
+
+   ```bash
+   yum clean all
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Clean shouldn't be needed.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 3]
```diff
@@ -61,19 +61,29 @@ The software upgrade process requires super-user privileges to run. Login as the
...
-   Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.
+Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.
 
 1. Run `chroma-config setup` to complete the installation.
 
-1. Restart the manager node.
+1. Update the repos on each server node (run from the manager node):
+
+   ```bash
+   iml update-repo <node name>
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: This is going to be using hostlist expression for the nodes.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 4]
```diff
@@ -144,10 +154,20 @@ IML requires that the filesystem(s) associated with each node to be upgraded mus
...
 
-In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. Next, begin the upgrade.
+In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. You may need to disable the CentOS base repo and enable the vault repos. For example:
+
+```bash
+sed -i "s/enabled=[0-1]//g" /etc/yum.repos.d/CentOS-Base.repo
+sed -i "s/gpgcheck=\([0-1]\)/gpgcheck=\1\nenabled=0/g" /etc/yum.repos.d/CentOS-Base.repo
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/os/x86_64/
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/extras/x86_64/
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/updates/x86_64/
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: I think we can skip this. If we target Lustre 2.12.3, it will support EL 7.7

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 5]
```diff
@@ -144,10 +154,20 @@ IML requires that the filesystem(s) associated with each node to be upgraded mus
...
+```
+
+Next, from the manager node, upgrade the OS for each host using a pdsh expression (you may need to install pdsh via yum).
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]:
```suggestion
Next, from the manager node, upgrade the OS for each host using a hostlist expression (you may need to install pdsh via yum).
```

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md [hunk 6]
```diff
@@ -144,10 +154,20 @@ IML requires that the filesystem(s) associated with each node to be upgraded mus
...
 
 ```bash
-yum -y upgrade --exclude=python2-iml*
+ pdsh -w root@mds[1,2] -w root@oss[1,2] "yum -y upgrade --exclude=python2-iml*"
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: We should leave out suggestions like pdsh I think and leave that up to the upgraders. Just stick to the upgrade steps directly.

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 7]
```diff
@@ -0,0 +1,186 @@
+# Upgrading Integrated Manager for Lustre 5.0.0 to Lustre {{site.lustre_version}} and Integrated Manager for Lustre {{site.version}}
+
+[**Upgrade Guide**](ug_TOC.md)
+
+## Upgrade Integrated Manager for Lustre
+
+The first component in the environment to upgrade is the Integrated Manager for Lustre server and software. The manager server upgrade can be conducted without any impact to the Lustre file system services.
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]:
```suggestion
The first component in the environment to upgrade is the Integrated Manager for Lustre server and software. The manager server upgrade can be conducted without any impact to the Lustre filesystem services.
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+1. Download the latest Integrated Manager for Lustre release repo:
+
+   ```sh
+   yum-config-manager --add-repo=https://github.com/whamcloud/integrated-manager-for-lustre/releases/download/v5.1.0/chroma_support.repo
+   ```
+
+1. Verify that the old iml-5.0 repo file has been removed from the repolist and that the 5.1 repo has been added.
+
+   ```bash
+   yum clean all
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]:
```suggestion
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+   yum repolist
+   ```
+
+1. Run the OS upgrade.
+
+   ```bash
+   yum clean all
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]:
```suggestion
   yum clean metadata
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+   ```
+
+Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.
+
+1. Run `chroma-config setup` to complete the installation.
+
+1. Update the repos on each server node (run from the manager node):
+
+   ```bash
+   iml update-repo <node name>
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: This is going to be using hostlist expression for the nodes.

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 11]
```diff
@@ -0,0 +1,186 @@
...
+## Upgrade the OS on each server node
+
+In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. You may need to disable the CentOS base repo and enable the vault repos. For example:
+
+```bash
+sed -i "s/enabled=[0-1]//g" /etc/yum.repos.d/CentOS-Base.repo
+sed -i "s/gpgcheck=\([0-1]\)/gpgcheck=\1\nenabled=0/g" /etc/yum.repos.d/CentOS-Base.repo
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/os/x86_64/
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/extras/x86_64/
+yum-config-manager --add-repo=http://vault.centos.org/7.6.1810/updates/x86_64/
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: I think we can skip this. If we target Lustre 2.12.3, it will support EL 7.7

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+```
+
+Next, from the manager node, upgrade the OS for each host using a pdsh expression (you may need to install pdsh via yum).
+
+```bash
+ pdsh -w root@mds[1,2] -w root@oss[1,2] "yum -y upgrade --exclude=python2-iml*"
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: We should leave out suggestions like pdsh I think and leave that up to the upgraders. Just stick to the upgrade steps directly.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 4 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: Sure, I'll revert.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md (hunk 6 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.0-el7.md]: sure, i'll update to:
```
yum -y upgrade --exclude=python2-iml*
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md (hunk 10 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: Updated to:
```
iml update-repo mds[1,2],oss[1,2]
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md (hunk 11 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: Sure, removed.

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md (hunk 12 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: Updated

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+   ```
+
+Refer to the operating system documentation for details on the correct procedure for upgrading between minor OS releases. Note that a mapping of new installs and upgrades will be displayed. Look through this chart carefully and verify that python2-iml-manager is marked for upgrade and that it will be upgraded to {{site.version}}.
+
+1. Run `chroma-config setup` to complete the installation.
+
+1. Update the repos on each server node (run from the manager node):
+
+   ```bash
+   iml update_repo --hosts mds[1,2],oss[1,2]
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: This is probably too specific. We should at least have a note that this an example and not canonical.

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+
+1. Navigate to _Configuration->Filesystems_
+1. For each filesystem listed:
+
+   1. Click the filesystem's `Actions` button
+   1. Select _Stop_
+
+## Upgrade the OS on each server node
+
+In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. Next, from the manager node, upgrade the OS for each host:
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]:
```suggestion
In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.7 packages. Next, from the manager node, upgrade the OS for each host:
```

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 15]
```diff
@@ -0,0 +1,175 @@
...
+
+## Stopping the filesystem
+
+IML requires that the filesystem(s) associated with each node to be upgraded must be stopped. Follow these steps:
+
+1. Navigate to _Configuration->Filesystems_
+1. For each filesystem listed:
+
+   1. Click the filesystem's `Actions` button
+   1. Select _Stop_
```
Comment (jgrund) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: We should stop the fs prior to updating the manager.

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md (hunk 13 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: updated to:
"Update the repos on each server node. As an example, consider the following hosts: mds1.local, mds2.local, oss1.local, and oss2.local:"

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md (hunk 15 above)
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]: I moved this section just before the update manager step.

File: docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.1-el7.md [hunk 16]
```diff
@@ -85,70 +98,68 @@ The upgrade procedure documented here describes the faster and more reliable app
...
-In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.6 packages. Next, begin the upgrade.
+In order to upgrade, make sure yum is configured on each server node to pull down CentOS 7.7 packages. Next, from the manager node, upgrade the OS for each host:
 
-```bash
-yum -y upgrade --exclude=python2-iml*
-```
+1. Upgrade the OS
+
+   ```bash
+   yum -y upgrade
```
[the remark this answers is not in the document]
Comment (johnsonw) [reply docs/Upgrade_Guide/upgrade_iml-4.0-el7_to_iml-5.1-el7.md]: agreed. Will update

File: docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md [hunk 17]
```diff
@@ -0,0 +1,181 @@
...
+   **Note:** This is not intended to be a comprehensive backup of the entire operating system configuration. It covers the essential components pertinent to Lustre servers managed by Integrated Manager for Lustre that are difficult to re-create if deleted. Make sure to backup any other important configuration files that may be on your system, such as multipath configurations.
+
+   The following files will need to be backed up if multipath is being used on the system:
+
+   ```sh
+   /etc/multipath/*
+   /etc/multipath.conf
+   ```
+
+1. Copy the backups for each server's configuration to a safe location tha
```
Comment (johnsonw) [inline docs/Upgrade_Guide/upgrade_iml-5.0-el7_to_iml-5.1-el7.md]:
```suggestion
1. Copy the backups for each server's configuration to a safe location that is not on the server being upgraded.
```
